src/components/step-flow/step-flow.spec.tsx

```diff
@@ -0,0 +1,361 @@
+              titleRef: mockRef,
+            }}
+          />
+        );
+
+        expect(
+          container.querySelector('[data-element="step-label"]')
+        ).toHaveTextContent(`${currentStep} of ${totalSteps}`);
+      }
+    );
+
+    it.each(generateCurrentStepOverTotalStepsVariations())(
+      "when 'totalSteps' is passed as %s and is lower than the 'currentStep' prop of %s, the step label contains the correct text",
+      (totalSteps, currentStep) => {
+        const { container } = render(
+          <StepFlow
+            {...{
+              title: "baz",
+              totalSteps,
+              currentStep,
+              titleTabIndex: 0,
+              titleRef: mockRef,
+            }}
+          />
+        );
+
+        expect(
+          container.querySelector('[data-element="step-label"]')
+        ).toHaveTextContent(`${totalSteps} of ${totalSteps}`);
+      }
+    );
+
+    it.each(generateLimitedVariations())(
+      "when 'totalSteps' is passed as %s and 'currentStep' prop is %s, the visually hidden title text contains the correct text",
+      (totalSteps, currentStep) => {
+        const category = "foo";
+        const title = "bar";
+
+        const { container } = render(
+          <StepFlow
+            {...{
+              category,
+              title,
+              totalSteps,
+              currentStep,
+              titleTabIndex: 0,
+              titleRef: mockRef,
+            }}
+          />
+        );
+
+        expect(
+          container.querySelector('[data-element="visually-hidden-title-text"]')
+        ).toHaveTextContent(
+          `${category}. ${title}. Step ${currentStep} of ${totalSteps}.`
+        );
+      }
+    );
+
+    it.each(generateCurrentStepOverTotalStepsVariations())(
+      "when 'totalSteps' is passed as %s and is lower than the 'currentStep' prop of %s, the visually hidden title text contains the correct text",
+      (totalSteps, currentStep) => {
+        const category = "foo";
+        const title = "bar";
+
+        const { container } = render(
+          <StepFlow
+            {...{
+              category,
+              title,
+              totalSteps,
+              currentStep,
+              titleTabIndex: 0,
+              titleRef: mockRef,
+            }}
+          />
+        );
+
+        expect(
+          container.querySelector('[data-element="visually-hidden-title-text"]')
+        ).toHaveTextContent(
+          `${category}. ${title}. Step ${totalSteps} of ${totalSteps}.`
+        );
+      }
+    );
+
+    it("when the 'showCloseIcon' prop is true, the correct element renders", () => {
+      const { container } = render(
+        <StepFlow
+          {...{
+            title: "baz",
+            totalSteps: 6,
+            currentStep: 1,
+            showCloseIcon: true,
+            titleRef: mockRef,
+          }}
+        />
+      );
+
+      expect(
+        container.querySelector('[data-element="close"]')
```

Comment:
comment: if we set and aria-label (see above) we can get by label here instead of using a container to query.